lib/beetle/message.rb

```diff
@@ -168,7 +170,7 @@ def set_timeout!
 
     # handler timed out?
     def timed_out?
```

Comment:
This method seems to be never called. The actual timeout check in the `process_internal` method directly reads the timeout key from Redis and checks it like [this](https://github.com/xing/beetle/blob/e01a56af285c4fb7655462e6f8d043bbd45679c3/lib/beetle/message.rb#L314) `!(timeout && timeout.to_i < now)`, thus `TIMEOUT_GRACE_PERIOD` is not considered. I suggest having a single method to check timeout and use it everywhere for checking timeout.

Reply:
good catch!

Reply:
yet we don't want access the dedup store again as we already have retrieved the timeout value.

Reply:
pushed new version